conda/meta.yaml

```diff
@@ -1,4 +1,4 @@
-{% set version = "3.0.6.dev3" %}
+{% set version = "3.0.6.dev4" %}
```

Comment:
Just in case it's a hassle for you to change this manually. Now that we have the version within the file ``VERSION``, would it be handy to have a very simple python script that replaces this bit of version in the ``meta.yaml``, triggered by a GitHub action?

For example, something like the following, summarised by our friend GPT:
````
import re
version_file_path = 'VERSION'
yaml_file_path = 'meta.yaml'

with open(version_file_path, 'r') as file:
    version = file.read().strip()

with open(yaml_file_path, 'r') as file:
    yaml_content = file.read()

new_yaml_content = re.sub(r'{% set version = ".*" %}', f'{{% set version = "{version}" %}}', yaml_content)

with open(yaml_file_path, 'w') as file:
    file.write(new_yaml_content)
````

Reply:
Again, only if it's less of a hassle to keep in mind that the VERSION is also in the YAML file

Reply:
Or actually, rather than modify this YAML file, why not have the other files read the file ``meta.yaml`` instead of ``VERSION``? That way, the version exists only in one place. We could use a regex to find the version within this yaml file and extract it instead

Reply:
I think it's okay for now, but it's something to keep in mind for a future PR... we can decide the direction of information flow (i.e. from VERSION to YAML or vice versa) there.